src/lib/pdfconverter.cc

```diff
@@ -546,13 +546,26 @@ void PdfConverterPrivate::findLinks(QWebFrame * frame, QVector<QPair<QWebElement
 		uexternal  = PageObject::webPageToObject[frame->page()]->settings.useExternalLinks;
 	}
 	if (!ulocal && !uexternal) return;
-	foreach (const QWebElement & elm, frame->findAllElements("a")) {
+	foreach (QWebElement elm, frame->findAllElements("a")) {
 		QString n=elm.attribute("name");
 		if (n.isEmpty()) n=elm.attribute("ns0:name");
 		if (n.startsWith("__WKANCHOR_")) anchors[n] = elm;
 
 		QString h=elm.attribute("href");
-		if (h.isEmpty()) h=elm.attribute("ns0:href");
+		if (h.isEmpty()) {
+			h=elm.attribute("ns0:href");
+
+			//Internal links (anchor tags) are not linked correctly
+			//when the anchor's content is empty. This adds a space with
+			//a height of 0 when it is empty so that internal links
+			//are linked correctly.
+			if(elm.firstChild().isNull()){
```

Comment:
You make a good point about other tags being used as targets which I didn't think about. Here are the cases I can think of:

1. An ```a``` tag with either ```id``` or ```name``` attributes, empty ```href``` or ```ns0:href``` attributes, and no children.
2. Any other anchor tags (e.g. ```div, p, h1```) that have an ```id``` and no children. I can look up the full list of tags than can be used as an anchor.

Am I missing any cases?